crates/ruff_linter/src/rules/flake8_pyi/rules/redundant_numeric_union.rs

```diff
@@ -66,57 +82,189 @@ pub(crate) fn redundant_numeric_union(checker: &mut Checker, parameters: &Parame
     }
 }
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-enum Redundancy {
-    FloatComplex,
-    IntComplex,
-    IntFloat,
-}
-
-fn check_annotation(checker: &mut Checker, annotation: &Expr) {
-    let mut has_float = false;
-    let mut has_complex = false;
-    let mut has_int = false;
+fn check_annotation<'a>(checker: &mut Checker, annotation: &'a Expr) {
+    let mut numeric_flags = NumericFlags::empty();
 
```

Comment:
Instead of `has_*`, I've used `bitflags` (which are used in some other parts of the code already).
(They correspond naturally to the order of Complex > Float > Int)